Mathlib/Analysis/Convex/Cone/Pointed.lean

```diff
@@ -0,0 +1,188 @@
+
+## TODO
+
+- Rewrite proper cones using pointed cones.
+
+-/
+
+variable {𝕜 E F G : Type*}
+
+-- TODO: remove `prettyPrint := false` once #6833 is merged
+local notation3 (prettyPrint := false) "𝕜≥0" => {c : 𝕜 // 0 ≤ c}
+
+/-- A pointed cone is a submodule of a module with scalars restricted to being nonnegative. -/
+abbrev PointedCone (𝕜 E) [OrderedSemiring 𝕜] [AddCommMonoid E] [Module 𝕜 E] :=
+  Submodule {c : 𝕜 // 0 ≤ c} E
+
+namespace PointedCone
+
+open Function
+
+section Definitions
+
+variable [OrderedSemiring 𝕜]
+variable [AddCommMonoid E] [Module 𝕜 E]
+
+instance : Coe (PointedCone 𝕜 E) (ConvexCone 𝕜 E) where
+  coe := fun S => {
+    carrier := S
+    smul_mem' := fun c hc _ hx => S.smul_mem ⟨c, le_of_lt hc⟩ hx
+    add_mem' := fun _ hx _ hy => S.add_mem hx hy }
+
+theorem coe_injective : Injective ((↑) : PointedCone 𝕜 E → ConvexCone 𝕜 E) :=
+  fun _ _ => by simp
+
+@[simp]
+theorem coe_pointed (S : PointedCone 𝕜 E) : (S : ConvexCone 𝕜 E).Pointed := by
+  simp [ConvexCone.Pointed]
+
+@[ext]
+theorem ext {S T : PointedCone 𝕜 E} (h : ∀ x, x ∈ S ↔ x ∈ T) : S = T :=
+  SetLike.ext h
+
+instance instZero (S : PointedCone 𝕜 E) : Zero S :=
+  ⟨0, S.zero_mem⟩
+
+/-- The `PointedCone` constructed from a pointed `ConvexCone`. -/
+def _root_.ConvexCone.toPointedCone {S : ConvexCone 𝕜 E} (hS : S.Pointed) : PointedCone 𝕜 E where
+  carrier := S
+  add_mem' := fun hx hy => S.add_mem hx hy
+  zero_mem' := hS
+  smul_mem' := fun ⟨c, hc⟩ x hx => by
+    simp_rw [SetLike.mem_coe]
+    cases' eq_or_lt_of_le hc with hzero hpos
+    · unfold ConvexCone.Pointed at hS
+      convert hS
+      simp [← hzero]
+    · apply ConvexCone.smul_mem
+      convert hpos
+      exact hx
+
+@[simp]
+lemma _root_.ConvexCone.mem_toPointedCone {S : ConvexCone 𝕜 E} (hS : S.Pointed) (x : E) :
+    x ∈ S.toPointedCone hS ↔ x ∈ S :=
+  Iff.rfl
+
+@[simp]
+lemma _root_.ConvexCone.coe_toPointedCone {S : ConvexCone 𝕜 E} (hS : S.Pointed) :
+    (S.toPointedCone hS : Set E) = S :=
+  rfl
+
+end Definitions
+
+section Maps
+
+variable [OrderedSemiring 𝕜]
+variable [AddCommMonoid E] [Module 𝕜 E]
+variable [AddCommMonoid F] [Module 𝕜 F]
+variable [AddCommMonoid G] [Module 𝕜 G]
+
+/-!
+
+## Maps between pointed cones
+
+There is already a definition of maps between submodules, `Submodule.map`. In our case, these maps
+are induced from linear maps between the ambient modules that are linear over nonnegative scalars.
+Such maps are unlikely to be of any use in practice. So, we construct some API to define maps
+between pointed cones induced from linear maps between the ambient modules that are linear over
+*all* scalars.
+
+-/
+
+/-- The image of a pointed cone under a `𝕜`-linear map is a pointed cone. -/
+def map (f : E →ₗ[𝕜] F) (S : PointedCone 𝕜 E) : PointedCone 𝕜 F :=
+  Submodule.map (f : E →ₗ[𝕜≥0] F) S
+
+@[simp, norm_cast]
+theorem coe_map (S : PointedCone 𝕜 E) (f : E →ₗ[𝕜] F) : (S.map f : Set F) = f '' S :=
+  rfl
```

Comment:
Can you add `toConvexCone_map` too?

Reply:
While writing this I discovered that `ConvexCone.map` was unnecessarily requiring `k` to be a `LinearOrderedField` while only `OrderedSemiring` is sufficient.